scrunch/expressions.py

```diff
@@ -526,7 +527,8 @@ def category_ids(var_id, var_value, variables=variables):
         if len(subitems) == 2:
             if 'value' in subitems[1] and 'variable' in subitems[0]:
                 var_url = subitems[0]['variable']
-                if ds.variables.index[var_url]['type'] == 'multiple_response':
+                var_index = ds.variables.index
```

Comment:
Note that you're still calling `.index` twice in the same code path first here and then inside inside `adapt_multiple_response` a few lines later here https://github.com/Crunch-io/scrunch/pull/141/files#diff-54e01706c90fd9d7e26c9f5f4d422b75R452.

The problem begins because you're using the variable URL to refer to it. You don't have the variable resource from before anywhere? It looks that we're reaching this point because somewhere before we could not get the variable shoji resource. You should be able to do `var_by_alias = ds.variables.by('alias')` at the top of the `process_expr` call and then use that inside all the loops you need instead of doing 2 GETs for each variable we want to validate in the expression. Then pass around that `var_by_alias` to the functions that need so you don't make more requests to get the same that you just got a few miliseconds ago.

After that change you will just be able to check if `variable.body.type == "multiple_response"` and avoid the double condition that looks non obvious (you're checking if the variable is in the index in case it is a subvariable).

You should avoid dealing with variable URLs as much as possible, I believe that most of this logic can be done without having to know the variable's URL,